assets/aml-benchmark/components/src/dataset_preprocessor/dataset_preprocessor.py

```diff
@@ -0,0 +1,136 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ---------------------------------------------------------
+
+"""DataPreprocessor class and runner."""
+
+import json
+import os
+import re
+import jinja2
+import sys
+from utils.exceptions import BenchmarkValidationException
+from utils.error_definitions import BenchmarkValidationError
+from utils.logging import get_logger
+from utils.io import read_jsonl_files
+from azureml._common._error_definition.azureml_error import AzureMLError
+
+current_folder = os.path.dirname(os.path.abspath(__file__))
+parent_folder = os.path.dirname(current_folder)
+sys.path.append(parent_folder)
+logger = get_logger(__name__)
+
+jinja2.filters.FILTERS['zip'] = zip
+ENV = jinja2.Environment()
+ENV.globals.update(zip=zip)
+
+
+class DatasetPreprocessor(object):
+    """DatasetPrerprocessor object class."""
+
+    def __init__(
+        self,
+        input_dataset: str = None,
+        template: str = None,
+        user_preprocessor: str = None,
+        encoder_config: str = None,
+        output_dataset: str = None
+    ):
+        """Dataset Preprocessor Class.
+
+        :param input_dataset: Path to the directory to load the dataset.
+        :param template: A json dictionary where key is the name of the column enclosed in " " and associated \
+            dict value is presented using jinja template logic which will be used to extract the \
+            respective value from the dataset.
+        :param user_preprocessor: Path to the custom preprocessor python script provided by user.
+        :param encoder_config: Dictionary in json format that contains idtolabel or labeltoid to encode. This is not \
+            applicable to custom scripts.
+        :param output_dataset: Path to the dump the processed .jsonl file.
+        """
+        self.input_dataset = input_dataset
+        self.template = template
+        self.user_preprocessor = user_preprocessor
+        self.encoder_config = encoder_config
+        self.output_dataset = output_dataset
+        self.validate()
+
+    def validate(self):

```

Comment:
Little bit of fun python: you can put this into `__post_init__` method. Then you don't need to explicitly call it, and it will be called for child classes too unless they override. 

```python
def __post_init__(self) -> None:
        self._validate()
```

Reply:
I can try this. Thank you.